### DISCUSSION TITLE
Is there a way to use floats in NumberInput without precision (atlernativly remove trailing zeros)?

### DISCUSSION THREAD

**Discussion by PointSingularity** [mantinedev/mantine#2541] [Q&A]
Here is my code:
```tsx
<NumberInput
  size="sm"
  type="text"
  decimalSeparator=","
  hideControls
  withAsterisk
  label={'Diameter'}
/>
```
If I use this for this input
![image](https://user-images.githubusercontent.com/56246797/192006577-a4611e3f-2b08-4796-b155-59c0f448360b.png)
after blur it goes to
![image](https://user-images.githubusercontent.com/56246797/192006638-21939827-0f76-47eb-b1f5-2de593ffb266.png)

But if I use precision, there are a lot of trailing zeros:
![image](https://user-images.githubusercontent.com/56246797/192007872-a597c04f-b26c-4b6d-997f-02b56bc6efe1.png)

Basically my question is can we have the best of both worlds, where we would type '225,23' and it would stay that way, without using precision, but still having the default behavior of only accepting numbers and the decimal separator as valid input?

**Comment by rtivital** [comment]
`precision` prop controls the number of digit displayed after decimal separator, if you do not want trailing zeros, then you need to reduce `precision` – https://codesandbox.io/s/serene-chebyshev-m4y0v9?file=/src/App.tsx

**Reply by PointSingularity** [reply]
Hey thanks for your answer!
Sorry for the late reply, but I've been thinking about it and I think I have a proposal for an additional property for the NumberInput "removePrecisionTrailingZeros":

If precision is enabled and clampOnBlur is on, then check the string for any zeroes that appear on the end of the string, and remove them.
Additionally, check if the last character is a separator, and remove it too.

Would it be ok, I opened a PR, so you can see if it is something that you would have in Mantine?

**Reply by rtivital** [reply]
Okay, I'm fine with it
